From 8207b6cf6d82815addea06dfe2846440abf85619 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Fri, 12 Sep 2025 14:03:35 +0200
Subject: [PATCH] arm64: dts: st: use rtc as clock provider for WiFi/BT on
 stm32mp257f-dk

Add the "#clock-cells" property in the rtc node.
Add the "clocks" property to the WiFi/BT combo nodes.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I506d9d9c48a4ead81a6b3a828b020e56d1e8f0e1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/490105
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index e24154a0c2c9..e148f056b74f 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -188,6 +188,8 @@ sound {
 	wifi_pwrseq: wifi-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		reset-gpios = <&gpiog 8 GPIO_ACTIVE_LOW>;
+		clocks = <&rtc>;
+		clock-names = "ext_clock";
 	};
 };
 
@@ -650,6 +652,7 @@ typec_ep: endpoint {
 };
 
 &rtc {
+	#clock-cells = <0>;
 	st,lsco = <RTC_OUT2_RMP>;
 	pinctrl-0 = <&rtc_out2_rmp_pins_a>;
 	pinctrl-names = "default";
@@ -784,6 +787,8 @@ &usart1 {
 	bluetooth {
 		shutdown-gpios = <&gpiog 4 GPIO_ACTIVE_HIGH>;
 		compatible = "brcm,bcm43438-bt";
+		clocks = <&rtc>;
+		clock-names = "lpo";
 		max-speed = <2000000>;
 		vbat-supply = <&scmi_v3v3>;
 		vddio-supply = <&scmi_v3v3>;
-- 
2.53.0

